### PR TITLE
chore(docs): proxy

### PR DESCRIPTION
**Pull request by RikuVan**
First go at this. Improvement suggestions 🙏

### CODE REVIEW

File: docs/basic/proxy.mdx [hunk 1]
```diff
@@ -6,6 +6,8 @@ description: 'Create a proxy object.'
 
 # proxy
 
+The proxy tracks changes to the original object and all nested objects, notifying listeners when a new version of the object is created.
```
Comment (dai-shi) [inline docs/basic/proxy.mdx:9]: > created

Precisely speaking, object doesn't have to be created, but just modified. Like `++state.count` doesn't create a new object.

File: docs/basic/proxy.mdx [hunk 2]
```diff
@@ -21,3 +23,82 @@ setInterval(() => {
...
+```
+
+#### Gotchas
+
+You cannot descructure the object from proxy and subscribe to a property.
+
+```jsx
+// don't
+const { user } = proxy({ user: { name: 'Timo' } })
+subscibe(user, () => {...})
```
Comment (dai-shi) [inline docs/basic/proxy.mdx:70]: This still works. Maybe you want to point out something more specific.

File: docs/basic/proxy.mdx [hunk 3]
```diff
@@ -21,3 +23,82 @@ setInterval(() => {
...
+
+// do
+const state = proxy({ user: { name: 'Nina' } })
+subscibe(state.user, () => {...})
```
Comment (dai-shi) [inline docs/basic/proxy.mdx:74]: Do you mean `subscribe(state, () => {...})` so that we can detect `state.user = null`-kind-of mutations?

**Review by dai-shi — COMMENTED** [review]
Thanks for working on this. Added some comments.

File: docs/basic/proxy.mdx (hunk 2 above)
Comment (RikuVan) [reply docs/basic/proxy.mdx:70]: ok, I thought this didn't work 😮

**Review by dai-shi — APPROVED** [review]
Looks good. Thanks for your work!

**Pull request merged**